jvm/src/test/kotlin/app/cash/trifle/CertificateRequestTest.kt

```diff
@@ -0,0 +1,120 @@
+    assertTrue(certificateRequest.verify())
+  }
+
+  @Test
+  fun verifyIncorrectlySignedRequest() {
+    val ecSpec = ECGenParameterSpec("secp256r1")
+    val generator = KeyPairGenerator.getInstance("EC")
+    generator.initialize(ecSpec, SecureRandom())
+
+    val keyPair: KeyPair = generator.generateKeyPair()
+    val keyPair2: KeyPair = generator.generateKeyPair()
+    val outputStream = ByteArrayOutputStream()
+    val noopTrifleContentSigner = object: TrifleContentSigner() {
+      override fun subjectPublicKeyInfo(): SubjectPublicKeyInfo =
+        SubjectPublicKeyInfo.getInstance(keyPair.public.encoded)
+
+      override fun getAlgorithmIdentifier(): AlgorithmIdentifier = TrifleAlgorithmIdentifier.ECDSASha256AlgorithmIdentifier
+
+      override fun getOutputStream(): OutputStream = outputStream
+
+      override fun getSignature(): ByteArray {
+        val signature = Signature.getInstance(
+          DefaultSignatureNameFinder()
+            .getAlgorithmName(algorithmIdentifier)
+        )
+        signature.initSign(keyPair2.private)
+        signature.update(outputStream.toByteArray())
+        outputStream.reset()
+
+        return signature.sign()
+      }
+    }
+
+    val pkcs10Request = PKCS10CertificationRequestBuilder(
+      X500Name("CN=testName"),
+      noopTrifleContentSigner.subjectPublicKeyInfo()
+    ).build(noopTrifleContentSigner)
+    val certificateRequest = CertificateRequest.PKCS10Request(pkcs10Request.encoded.toByteString())
+
+    assertFalse(certificateRequest.verify())
+  }
+
+  @Test
+  fun verifyUnknownAlgorithm() {
+    // Choose an unsupported signing algorithm type.
+    val keyPair = KeyPairGenerator.getInstance("DSA").generateKeyPair();
+    val outputStream = ByteArrayOutputStream()
+    val unsupportedTrifleContentSigner = object: TrifleContentSigner() {
+      override fun subjectPublicKeyInfo(): SubjectPublicKeyInfo =
+        SubjectPublicKeyInfo.getInstance(keyPair.public.encoded)
+
+      override fun getAlgorithmIdentifier(): AlgorithmIdentifier = subjectPublicKeyInfo().algorithm
+
+      override fun getOutputStream(): OutputStream = outputStream
+
+      override fun getSignature(): ByteArray {
+        val signature = Signature.getInstance(
+          DefaultSignatureNameFinder()
+            .getAlgorithmName(algorithmIdentifier)
+        )
+        signature.initSign(keyPair.private)
+        signature.update(outputStream.toByteArray())
+        outputStream.reset()
+
+        return signature.sign()
+      }
+    }
+
+    val pkcs10Request = PKCS10CertificationRequestBuilder(
+      X500Name("CN=testName"),
+      unsupportedTrifleContentSigner.subjectPublicKeyInfo()
+    ).build(unsupportedTrifleContentSigner)
+    val certificateRequest = CertificateRequest.PKCS10Request(pkcs10Request.encoded.toByteString())
+    val exception = assertThrows<PKCSException> {
+      certificateRequest.verify()
+    }
+    assertEquals(
```

Comment:
instead of breaking out the assertThrows and checking the message body. consider:
```
assertThrows<ExceptionClass>(<expectedMessage) { executionBlock }
```

Reply:
much cleaner, yes.